tests/test_utils.py

```diff
@@ -0,0 +1,30 @@
+from dataclasses import dataclass
+
+from hypothesis import given, strategies as st
+
+from hypofuzz.utils import k_way_merge
+
+
+@given(st.lists(st.lists(st.integers(), min_size=1).map(sorted)))
+def test_k_way_merge(lists):
+    assert k_way_merge(lists) == sorted(sum(lists, []))
+
+
+@dataclass
+class A:
+    value: int
+
+
+def test_k_way_merge_key():
```

Comment:
Could just add `key=sampled_from([None, lambda x: x, lambda x: -x])` to the test above.